btc/guardian.go

```
@@ -0,0 +1,672 @@
+
+	signedTxBytes, err := hex.DecodeString(signTxResp.TxHex)
+	if err != nil {
+		return err
+	}
+
+	return tx.Deserialize(strings.NewReader(string(signedTxBytes)))
+}
+
+func (gc *GuardianClient) updateTx(prevTxID chainhash.Hash, updatedTx *wire.MsgTx, mergeTxs []*wire.MsgTx, values []int64) error {
+	var buf = new(bytes.Buffer)
+	updatedTx.Serialize(buf)
```

golangci-lint (1.62.2)

Error return value of `updatedTx.Serialize` is not checked

(errcheck)